src/context/analysisResultStateSlice.ts

```diff
@@ -0,0 +1,311 @@
+  checkBaselineDataLayer,
+  fetchApiData,
+  generateFeaturesFromApiData,
+  scaleAndFilterAggregateData,
+} from '../utils/analysis-utils';
+import { getWCSLayerUrl } from './layers/wms';
+import { getBoundaryLayerSingleton, LayerDefinitions } from '../config/utils';
+import { Extent } from '../components/MapView/Layers/raster-utils';
+import { layerDataSelector } from './mapStateSlice';
+import { LayerData, LayerDataParams, loadLayerData } from './layers/layer-data';
+import { DataRecord, NSOLayerData } from './layers/nso';
+import { BoundaryLayerData } from './layers/boundary';
+
+type AnalysisResultState = {
+  result?: AnalysisResult;
+  error?: string;
+  isLoading: boolean;
+  isMapLayerActive: boolean;
+};
+export class AnalysisResult {
+  key: number = Date.now();
+  featureCollection: FeatureCollection;
+  tableData: TableRow[];
+  rawApiData?: object[];
+  statistic: AggregationOperations;
+  legend: LegendDefinition;
+  hazardLayerId: WMSLayerProps['id'];
+  baselineLayerId: NSOLayerProps['id'];
+
+  constructor(
+    tableData: TableRow[],
+    featureCollection: FeatureCollection,
+    hazardLayer: WMSLayerProps,
+    baselineLayer: NSOLayerProps,
+    statistic: AggregationOperations,
+    rawApiData?: object[],
+  ) {
+    this.featureCollection = featureCollection;
+    this.tableData = tableData;
+    this.statistic = statistic;
+    this.legend = baselineLayer.legend;
+    this.rawApiData = rawApiData;
+
+    this.hazardLayerId = hazardLayer.id;
+    this.baselineLayerId = baselineLayer.id;
+  }
+  getHazardLayer(): WMSLayerProps {
+    return LayerDefinitions[this.hazardLayerId] as WMSLayerProps;
+  }
+  getBaselineLayer(): NSOLayerProps {
+    return LayerDefinitions[this.baselineLayerId] as NSOLayerProps;
+  }
+}
+
+export type TableRow = {
+  localName: string;
+  name: string;
+  baselineValue: DataRecord['value'];
+  coordinates?: Position;
+} & { [k in AggregationOperations]?: number };
+
+const initialState: AnalysisResultState = {
+  isLoading: false,
+  isMapLayerActive: true, // TODO toggle via UI
+};
+
+function generateTableFromApiData(
+  statistic: AggregationOperations,
+  aggregateData: AsyncReturnType<typeof fetchApiData>, // data from api
+  // admin layer, both props and data. Aimed to closely match LayerData<BoundaryLayerProps>
+  {
+    layer: adminLayer,
+    data: adminLayerData,
+  }: { layer: BoundaryLayerProps; data: BoundaryLayerData },
+  // baseline layer, both props and data
+  {
+    layer: { adminLevel },
+    data: { layerData },
+  }: { layer: NSOLayerProps; data: NSOLayerData },
+): TableRow[] {
+  // find the key that will let us reference the names of the bounding boxes.
+  const adminLevelName =
+    adminLayer.adminLevelNames[adminLevel - 1] || adminLayer.adminLevelNames[0];
+  // for local name too.
+  const adminLevelLocalName =
+    adminLayer.adminLevelLocalNames[adminLevel - 1] ||
+    adminLayer.adminLevelLocalNames[0];
+
+  return aggregateData.map((row: any) => {
```

Comment:
We can a little...
This typing is due to aggregateData being from an API request and hence being typed as `Array<object>`
To improve this we could possibly type aggregateData as `{[k in string]: string|number}` but that's likely the best we can do unless we have a solid typing for the API request.

Reply:
Solid typing for the API requests is probably a good idea. But if it adds too much work, don't worry about it.

Reply:
Doesn't seem too straightforward yet since it uses pseudo-names from the layers config as keys + some other fields I'm not too sure of. Eric might be helpful here, but from my rough verdict it won't be easy. I'm currently trying to piece together any future things the UI might need in the future for an issue; I'll make sure to mention and tag it here so its recorded :+1: